### PR TITLE
feat: Allow kzip protobuf encoding to be configurable

### PR DESCRIPTION
**Pull request by Arm1stice**
The popular third-party Rust protobuf crates do not support reading json-type protobufs. Therefore, the Rust extractor needs to be able to create kzips that use the proto format. This PR makes that a configurable option. It also enables this new option on the Rust extractor.

### CODE REVIEW

File: kythe/go/extractors/bazel/extractor.go [hunk 1]
```diff
@@ -68,10 +68,11 @@ import (
...
 type Config struct {
-	Corpus   string          // the default corpus label to use
-	Language string          // the language label to apply
-	Rules    vnameutil.Rules // rules for rewriting file VNames
-	Verbose  bool            // whether to emit verbose (per-file) logging
+	Corpus      string          // the default corpus label to use
+	Language    string          // the language label to apply
+	Rules       vnameutil.Rules // rules for rewriting file VNames
+	Verbose     bool            // whether to emit verbose (per-file) logging
+	ProtoFormat string          // output format of compilation unit (json or proto)
```
Comment (jaysachs) [inline kythe/go/extractors/bazel/extractor.go:75]: Why use a string here and not the enum?

File: kythe/go/extractors/bazel/settings.go [hunk 2]
```diff
@@ -74,6 +74,8 @@ func (s *Settings) SetFlags(f *flag.FlagSet, prefix string) func() {
 		"Only match source paths within the target package")
 	f.BoolVar(&s.Verbose, p("verbose"), false,
 		"Enable verbose (per-file) logging")
+	f.StringVar(&s.ProtoFormat, p("proto_format"), "json",
```
Comment (jaysachs) [inline kythe/go/extractors/bazel/settings.go:77]: There's https://github.com/kythe/kythe/blob/master/kythe/go/platform/tools/kzip/flags/encoding.go ... maybe it needs a more central location first.

File: kythe/go/extractors/cmd/bazel/extract_kzip/extract_kzip.go [hunk 3]
```diff
@@ -25,7 +25,6 @@ import (
 	"time"
 
 	"kythe.io/kythe/go/extractors/bazel"
-	"kythe.io/kythe/go/extractors/bazel/extutil"
```
Comment (jaysachs) [inline kythe/go/extractors/cmd/bazel/extract_kzip/extract_kzip.go:28]: Is there a corresponding BUILD dependency to prune?

Comment (Arm1stice) [reply kythe/go/extractors/cmd/bazel/extract_kzip/extract_kzip.go:28]: Ah you're right. Removed unused dependency

File: kythe/go/extractors/bazel/extractor.go (hunk 1 above)
Comment (Arm1stice) [reply kythe/go/extractors/bazel/extractor.go]: Good point. Changed this to use `kzip.Encoding`

Comment (justbuchanan) [reply kythe/go/extractors/bazel/extractor.go]: I'd rename the field to something like "KzipEncoding" over "ProtoFormat".

Comment (Arm1stice) [reply kythe/go/extractors/bazel/extractor.go]: Renamed fields

**Comment by Arm1stice** [comment]
After internal discussions, closing this in favor of #4547

**Pull request closed without merging**